code/app/src/main/java/com/example/qrazyqrsrus/Event.java

```diff
@@ -314,16 +321,34 @@ public void setCheckIns(ArrayList<String> checkIns) {
      *
      * @return
      */
-    public void addCheckIn(String checkIn) {
+    public boolean addCheckIn(String checkIn) {
+        if (this.maxAttendees != null && this.getAttendeeCount() >= this.maxAttendees) {
```

Comment:
This would stop people who signed up before the maxAttendees was reached to be unable to check in